CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc

```diff
@@ -8,13 +8,16 @@ static const double cap = 6.0;
 static const double tspar0[2] = {15.5, 12.2999};
 static const double tspar1[2] = {-3.2,-2.19142};
 static const double tspar2[2] = {32, 0};
+static const double tspar0_siPM[2] = {0, 0};
+static const double tspar1_siPM[2] = {0, 0};
+static const double tspar2_siPM[2] = {0, 0};
 
 double HcalTimeSlew::delay(double fC, BiasSetting bias) {
   double rawDelay=tzero[bias]+slope[bias]*log(fC);
   return (rawDelay<0)?(0):((rawDelay>tmax[bias])?(tmax[bias]):(rawDelay));			   
 }
 
-double HcalTimeSlew::delay(double fC, ParaSource source, BiasSetting bias, double par0, double par1, double par2) {
+double HcalTimeSlew::delay(double fC, ParaSource source, BiasSetting bias, double par0, double par1, double par2, bool isHPD) {
```

Comment:
@davidlange6 
please clarify what you mean.
Both HPD and SiPM hits are in the same input collection.

Another alternative is to create two instances of the hit processing code configured appropriately and then swap out the appropriate one at the right point.
There is no single point of entry for all hit processing algorithms. So, percolation of this alternative will need some refactoring.
Is this the way you suggest to go?

@igv4321